server.py

```diff
@@ -150,9 +170,73 @@ def rankings():
     return template("rankings.html", people=order)
 
 
+def logggedIn():
+    if not request.get_cookie("s_id"):
+        return False
+    with shelve.open(sessions_db) as sessions:
+        if not request.get_cookie("s_id") in sessions:
+            return False
+    return True
+
+
+def createSession(username):
+    session_id = "".join(
+        random.choice(string.ascii_letters + string.digits) for i in range(20)
+    )
+    response.set_cookie(
+        "s_id",
+        session_id,
+        expires=datetime.datetime.now() + datetime.timedelta(days=30),
+    )
+    with shelve.open(sessions_db) as sessions:
+        sessions[session_id] = username
+    return redirect("/dashboard")
+
+
+@app.post("/login")
+def login():
+    username = request.forms.get("username")
+    password = request.forms.get("password")
+    with shelve.open(user_db) as users:
+        if not username in users:
+            return "user does not exist."
```

Comment:
let's use an `abort` here

Reply:
`return abort(404, 'user does not exist')`

Reply:
I was thinking of sending the user back to login page and `user does not exist will be shown ` on the same page, that's why I just returned the text here and left if for later.

Reply:
ah. makes sense. Let's do that then

Reply:
I will code it and then you can review